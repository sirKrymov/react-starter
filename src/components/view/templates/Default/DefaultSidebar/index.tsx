// @SETUP: Please don't forget to remove this component or current comment if component is used in project
// @TODO: Add component tests

export function DefaultSidebar() {
  return;
}
