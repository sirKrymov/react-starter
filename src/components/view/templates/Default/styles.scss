.default-template {
  background-color: var(--bgPrimary);
  min-height: 100vh;
  position: relative;
  padding: 84px 24px;

  main {
    width: 100%;
  }
}
