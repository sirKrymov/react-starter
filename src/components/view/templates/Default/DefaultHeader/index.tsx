// @SETUP: Please don't forget to remove this component or current comment if component is used in project
// @TODO: Add component tests
import React, { ReactElement } from 'react';

import { AccountInfoContainer } from 'components/containers/AccountInfo';
import { ContentLimiter } from 'components/view/layouts/ContentLimiter';
import { LngChanger } from 'components/view/compound/LngChanger';
import { Logo } from 'components/view/atoms/Logo';

import './styles.scss';

export function DefaultHeader(): ReactElement {
  return (
    <header className="default-header">
      <ContentLimiter>
        <div className="default-header__content">
          <Logo id="default-logo" />

          <div style={{ display: 'flex', alignItems: 'center' }}>
            <LngChanger />

            <AccountInfoContainer />
          </div>
        </div>
      </ContentLimiter>
    </header>
  );
}
