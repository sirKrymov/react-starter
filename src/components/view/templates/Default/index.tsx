// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

import cn from 'classnames';

import { DefaultHeader } from './DefaultHeader';

import './styles.scss';

interface IProps {
  className?: string;
  children: ReactNode;
}

export function DefaultTemplate({ className, children }: IProps): ReactElement {
  const defaultTemplateClass = cn(
    {
      'default-template': true
    },
    className
  );

  return (
    <div className={defaultTemplateClass}>
      <DefaultHeader />

      <main>{children}</main>
    </div>
  );
}
