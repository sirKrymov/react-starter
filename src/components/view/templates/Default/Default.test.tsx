import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';

import { DefaultTemplate } from '../Default';

describe('<Default> template component', () => {
  let shallowComponent: ShallowWrapper;

  const createShallowComponent = () => {
    shallowComponent = shallow(
      <DefaultTemplate>
        <div>Test</div>
      </DefaultTemplate>
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createShallowComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowComponent).toMatchSnapshot();
    });
  });
});
