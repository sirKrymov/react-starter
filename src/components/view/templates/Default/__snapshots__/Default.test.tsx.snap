// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Default> template component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="default-template"
>
  <DefaultHeader />
  <main>
    <div>
      Test
    </div>
  </main>
</div>
`;
